src/app/Tests/AccountControllerTests.cs

```diff
@@ -0,0 +1,614 @@
+
+            var controller = CreateInstance();
+            controller.ControllerContext = controllerContext;
+
+            // Act
+            var result = await controller.LogoutAsync();
+
+            // Assert
+            _authService.Verify(service => service.UnsetTokensAsync(currentUserId), Times.Once);
+            Assert.IsAssignableFrom<OkResult>(result);
+        }
+
+        [Fact]
+        public void GetOAuthUrlShouldReturnRedirect()
+        {
+            // Arrange
+            var url = "test.test";
+
+            _googleService.Setup(service => service.GetOAuthUrl())
+                .Returns(url);
+
+            var controller = CreateInstance();
+
+            // Act
+            var result = controller.GetOAuthUrl();
+
+            // Assert
+            Assert.True(result is RedirectResult redirectResult && redirectResult.Url == url);
+        }
+
+        [Fact]
+        public async void SignInGoogleWithCodeShouldReturnNotFoundWhenPayloadIsNull()
+        {
+            // Arrange
+            var model = new SignInGoogleModel {Code = "test code"};
+            _googleService.Setup(service => service.ExchangeCodeForTokenAsync(model.Code))
+                .ReturnsAsync((GooglePayloadModel)null);
+
+            var controller = CreateInstance();
+
+            // Act
+            var result = await controller.SignInGoogleWithCodeAsync(model);
+
+            // Assert
+            Assert.IsAssignableFrom<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public async void SignInGoogleWithCodeShouldCreateUserWhenItDoesNotExist()
+        {
+            // Arrange
+            var model = new SignInGoogleModel { Code = "test code" };
+            var payload = new GooglePayloadModel
+            {
+                Email = "test@test.com",
+                GivenName = "Test",
+                FamilyName = "Tester"
+            };
+
+            _userService.Setup(service => service.FindByEmailAsync(payload.Email))
+                .ReturnsAsync((User)null);
+
+            _userService.Setup(service => service.CreateUserAccountAsync(It.IsAny<CreateUserGoogleModel>()))
+                .ReturnsAsync(new User());
+
+            _googleService.Setup(service => service.ExchangeCodeForTokenAsync(model.Code))
+                .ReturnsAsync(payload);
+
+            var controller = CreateInstance();
+
+            // Act
+            await controller.SignInGoogleWithCodeAsync(model);
+
+            // Assert
+            _userService.Verify(service => service.CreateUserAccountAsync(
+                It.Is<CreateUserGoogleModel>(m => m.Email == payload.Email &&  m.FirstName == payload.GivenName && m.LastName == payload.FamilyName)),
+                Times.Once
+            );
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async void SignInGoogleWithCodeShouldShouldEnableGoogleAuthWhenItDoesNotEnabled(bool isEnabled)
```

Comment:
The test is not passed:
`
Expected invocation on the mock should never have been performed, but was 1 times: service => service.CreateUserAccountAsync(It.IsAny<CreateUserGoogleModel>())
`

Reply:
@yarmolovichalex, Will be passed after google oauth is fixed